sources/@roots/bud-dashboard/src/components/index.tsx

```diff
@@ -1,17 +1,51 @@
-import type {Framework} from '@roots/bud-framework'
+import {Framework} from '@roots/bud-framework'
 import {useStdin} from 'ink'
-import React from 'react'
+import React, {useEffect, useState} from 'react'
+import {useEffectOnce, useUpdate} from 'react-use'
+import {StatsCompilation} from 'webpack'
 
 import {Input} from './input.component'
 import {Output} from './output.component'
 
 /**
- * Dashboard display component
+ * Build display component
  *
  * @public
  */
-export const Dashboard = ({tap}: {tap: () => Framework}) => {
+export const Build = ({tap}: {tap: () => Framework}) => {
```

Comment:
I moved all the logic around re-rendering into the root component.

This is a challenge because the rest of the framework is not encapsulated by react and so react doesn't respond to changes in its state.

In particular I need it to respond to changes to state held by `bud.compiler`. The most obvious solution is to pass `bud.compiler` in and initiate compilation here. But, that gets messy fast and ultimately you have to bring in `bud.server` as well. Then you have to dig through all the react stuff in the stack when something goes wrong. 👎🏼

I like this forced render strategy as a way to dodge all of that.